waiter/src/waiter/process_request.clj

```diff
@@ -484,16 +484,15 @@
 
 (defn wrap-descriptor
   "Adds the descriptor to the request/response.
-   Redirects users in the case of missing user/run-as-requestor."
+  Redirects users in the case of missing user/run-as-requestor."
   [handler request->descriptor-fn]
   (fn [request]
-    (try
-      (let [descriptor (request->descriptor-fn request)
-            handler (-> handler
-                        (middleware/wrap-context {:descriptor descriptor}))]
-        (handler request))
-      (catch Exception e
-        (if (missing-run-as-user? e)
+    (let [[descriptor error] (utils/tryv (request->descriptor-fn request))]
```

Comment:
I worry that `tryv` will lead to a lot of nested `(let [[... error] (tryv ...)] (if-not error ...))` blocks. Should we instead target a macro for that pattern where the macro would expect two arguments and behave similarly to an `if-let`?

Reply:
`try-let` doesn't support destructuring.  Do you still want to proceed with `try-let`?

Reply:
That is a `todo` item in `try-let`. We can have our own `try-let` macro if you like. My concern is with avoiding the extra if and I like `try-let` feature of avoiding the need for the vector.

Reply:
https://github.com/rufoa/try-let/pull/1 Let's see.